### ISSUE TITLE
jsonpatch regression in 0.982/1

### ISSUE THREAD

**Issue by henryiii** [python/mypy#13813]
**Crash Report**

MyPy can no longer import jsonpatch. No experimental flags, it just fails now.

**Traceback**

```
Deferral trace:
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
    jsonpatch:98
    jsonpatch:98
    jsonpatch:98
    jsonpatch:-1
    jsonpatch:99
    jsonpatch:99
    jsonpatch:99
    jsonpatch:-1
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:100
    jsonpatch:-1
    jsonpatch:102
    jsonpatch:102
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:101
    jsonpatch:-1
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:107
    jsonpatch:-1
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:112
    jsonpatch:-1
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:113
    jsonpatch:-1
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:114
    jsonpatch:236
    jsonpatch:237
.nox/check/lib/python3.10/site-packages/jsonpatch-stubs/__init__.pyi: error: INTERNAL ERROR: maximum semantic analysis iteration count reached
/Users/henryschreiner/git/teaching/level-up-your-python/.nox/check/lib/python3.10/site-packages/jsonpatch-stubs/__init__.pyi:229: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.982
Traceback (most recent call last):
  File "mypy/semanal.py", line 5915, in accept
  File "mypy/nodes.py", line 811, in accept
  File "mypy/semanal.py", line 801, in visit_func_def
  File "mypy/semanal.py", line 833, in analyze_func_def
  File "mypy/semanal.py", line 5602, in defer
AssertionError: Must not defer during final iteration
/Users/henryschreiner/git/teaching/level-up-your-python/.nox/check/lib/python3.10/site-packages/jsonpatch-stubs/__init__.pyi:229: : note: use --pdb to drop into pdb
```

**To Reproduce**

Type check this file:

```python
# tmp.py
import jsonpatch
```

Example run with nox:

```python
# noxfile.py
import nox
from pathlib import Path


@nox.session
def check(session: nox.Session) -> None:
    session.install("mypy", "types-jsonpatch")
    Path("tmp.py").write_text("import jsonpatch")
    session.run("mypy", "tmp.py", "--show-traceback")
```

```console
pipx run nox
```

**Your Environment**

- Mypy version used: 0.982 & 0.981
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.10
- Operating system and version: macOS. Also seen in CI on Linux in https://github.com/scikit-hep/pyhf/issues/2039

**Comment by henryiii** [comment]
Ahh, of course I open before trying master...

```python
import nox
from pathlib import Path


@nox.session
def check(session: nox.Session) -> None:
    session.install("mypy @ git+https://github.com/python/mypy@master", "types-jsonpatch")
    Path("tmp.py").write_text("import jsonpatch")
    session.run("mypy", "tmp.py", "--show-traceback")
```

Passes, while `@v0.982` fails. So I think this has been fixed.

**Comment by hauntsaninja** [comment]
Thanks for the report. This was fixed by #13516. We can't backport that, but you're welcome to use `--enable-recursive-aliases` if you need to unblock yourself.

**Comment by matthewfeickert** [comment]
> This was fixed by #13516.
Thanks!

> We can't backport that, but you're welcome to use `--enable-recursive-aliases` if you need to unblock yourself.

That's okay. We can just wait for the next release of `mypy` and have `pre-commit` pick that up to address https://github.com/scikit-hep/pyhf/issues/2039. :+1:

**Comment by henryiii** [comment]
> We can't backport that, but you're welcome to use --enable-recursive-aliases if you need to unblock yourself.

That's a reasonable workaround. Thanks!
